### PR TITLE
Use node 'require' instead of inlining scripts

### LINKED ISSUE #51: Figure out a way to not inline the scripts
I had to [inline the vendor scripts](https://github.com/sindresorhus/anatine/blob/2a6ae01d5df4886543225b6d66e87a1ff5ca988f/browser.js#L4-L18) as Electron only supports preloading one script. We need to figure out a better way or fix the Electron bug (https://github.com/electron/electron/issues/5400).

@paulmolluzzo @danhp Thoughts?

### PR DESCRIPTION
**Pull request by danhp**
Alternate to #56 
Requires `nodeIntegration` though.

Closes #51

### CODE REVIEW

File: browser.js [hunk 1]
```diff
@@ -292,16 +274,15 @@ function zoomInit() {
...
-	// TODO: Figure out how to loud external scripts here
-	// [
-	// 	path.resolve('vendor/mousetrap.js'),
-	// 	path.resolve('vendor/mousetrap-global-bind.js')
-	// ].forEach(src => {
-	// 	const script = document.createElement('script');
-	// 	script.src = `file://${src}`;
-	// 	script.async = false;
-	// 	document.head.appendChild(script);
-	// });
```
Comment (sindresorhus) [inline browser.js:304]: @danp Do you have any idea why the above didn't work? Maybe something to do with asar.

Maybe we could create our own custom protocol and load it through there. I think Atom does this. See: http://electron.atom.io/docs/api/protocol/

**Comment by paulmolluzzo** [comment]
Is there a reason we _don't_ want `nodeIntegration`? What was the motivation behind disabling it?

This looks to be a slight improvement over the other solution since there's no build process involved, it's just a direct load, but maybe there was a specific reason @sindresorhus had it commented it out?

File: browser.js (hunk 1 above)
Comment (paulmolluzzo) [inline browser.js:304]: @danhp

Comment (danhp) [inline browser.js:304]: Chrome doesn't allow webpages to load inline scripts in that way so they were just blocked.
image

Comment (sindresorhus) [inline browser.js:304]: Yeah, I think that's the reason Atom uses a custom protocol, it doesn't fall under the same restrictions.

**Comment by sindresorhus** [comment]
I had problems in Caprine with `nodeIntegration: true`. It caused the Facebook Messenger code to not work. If it works fine with Twitter, I think this would be the best solution.

File: index.js [hunk 2]
```diff
@@ -48,7 +48,6 @@ function createMainWindow() {
 		backgroundColor: storage.get('darkMode') ? '#192633 ' : '#fff',
 		webPreferences: {
 			preload: path.join(__dirname, 'browser.js'),
-			nodeIntegration: false,
```
Comment (sindresorhus) [inline index.js:51]: Comment it out instead with a comment about enabling it again when the Electron preload ticket is fixed.

File: browser.js [hunk 3]
```diff
@@ -292,16 +274,15 @@ function zoomInit() {
...
+	// load vendor scripts.
```
Comment (sindresorhus) [inline browser.js]: drop the `.`

File: browser.js [hunk 4]
```diff
@@ -1,23 +1,4 @@
...
-c);"function"===typeof define&&define.amd&&define(function(){return c})})(window,document);
-
-/* mousetrap-global-bind.js */
-(function(a){var c={},d=a.prototype.stopCallback;a.prototype.stopCallback=function(e,b,a,f){return this.paused?!0:c[a]||c[f]?!1:d.call(this,e,b,a)};a.prototype.bindGlobal=function(a,b,d){this.bind(a,b,d);if(a instanceof Array)for(b=0;b<a.length;b++)c[a[b]]=!0;else c[a]=!0};a.init()})(Mousetrap);
-
-/* eslint-enable */
-
 /*
 
 This is copied from:
```
Comment (sindresorhus) [inline browser.js]: The below should also be moved into a separate file.

File: browser.js [hunk 5]
```diff
@@ -292,16 +217,21 @@ function zoomInit() {
...
+	].forEach(src => {
+		const script = document.createElement('script');
+		script.textContent = `require('${src}')`;
+		document.head.appendChild(script);
+	});
+
+	// load scrollToTweet as a module.
+	const scrollToTweetPath = path.resolve('vendor/scroll-to-tweet.js');
+	const script = document.createElement('script');
+	script.textContent = `scrollToTweet = require('${scrollToTweetPath}')`;
```
Comment (danhp) [inline browser.js]: There's probably a better way of doing this...

Comment (sindresorhus) [inline browser.js]: Yes, just wrap the `scroll-to-tweet` in a require/browser wrapper:

``` js
(() => {
        // code here

    if (typeof module !== 'undefined' && module.exports) {
        module.exports = scrollToTweet;
    } else {
        window.scrollToTweet = scrollToTweet;
    }
})();
```

**Comment by danhp** [comment]
I think I got it this time!

File: vendor/scroll-to-tweet.js [hunk 6]
```diff
@@ -1,57 +1,59 @@
...
 
-		return tweetIsBelowNav(tweet.offsetTop);
-	});
+		window.scrollTo(0, scrollTarget);
+	};
 
-	window.scrollTo(0, scrollTarget);
-};
+	window.scrollToTweet = scrollToTweet;
+})();
```
Comment (paulmolluzzo) [inline vendor/scroll-to-tweet.js:59]: There is an extra empty lines at the end of this file. 😄

**Comment by paulmolluzzo** [comment]
This is looking good for me with the exception of that one extra empty line in `scroll-to-tweet.js`. All the functionality seems to be the same, and we're down to just 1 `TODO` remaining.

Nice work @danhp! 👍

**Comment by sindresorhus** [comment]
Awesome! Good solution :)

**Pull request merged**